modules/diskq/qdisk.c

```diff
@@ -544,30 +582,22 @@ _load_state(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *qoverflow)
     {
       if (!_load_non_reliable_queues(self, qout, qbacklog, qoverflow))
         return FALSE;
-    }
-
-  if (!self->options->read_only)
-    {
-      end_ofs = qout_ofs;
-      if (qbacklog_ofs && qbacklog_ofs < end_ofs)
-        end_ofs = qbacklog_ofs;
-      if (qoverflow_ofs && qoverflow_ofs < end_ofs)
-        end_ofs = qoverflow_ofs;
-      if(end_ofs > QDISK_RESERVED_SPACE)
-        _truncate_file(self, end_ofs);
-    }
 
-  if (!self->options->reliable)
-    {
-      self->file_size = qout_ofs;
-      _reset_queue_pointers(self);
+      gint64 end_ofs = QDISK_RESERVED_SPACE;
```

Comment:
note1: The old code did not truncate in case the file was "empty". check: Is it implemented elsewhere (in the event of "emptying" the queue)? I think this is a nice feature from the new one.

note2: _truncate_file returns with a bool, neither the old code, or the new one handles failure.

Reply:
About note1: Yes, truncating can happen in several cases in current implementation, e.g. during:
- popping messages from the queue (when the file becomes empty, only in non-rel. disk-queue)
- pushing messages to the queue (when there is unused space after write_head, i.e. read follows write)
- acking back messages from backlog (in case of reliable disk-queue where backlog is saved on the disk as well)
- opening, loading existing disk-queues (had a bug, which is fixed by this PR)

note2:
That's a good observation, indeed the result of the truncating is ignored in the entire disk-queue code.
It needs some thoughts to given, as it is not trivial how to resolve issues like that.
We have a restart mechanism (close current file as corrupted and open a new disk-queue file), which currently only used in case of a disk read operation fails during message popping.
The restart() mechanism is defined on LogQueueDisk level, we are on QDisk level.
We could propagate the error and see if we could either start a new disk-queue file or do something else.

Anyway, as you mentioned handling file truncating failures never handled, and I did not changed that either, just fixed a bug in one of the truncating scenarios.
I have made a lot of refactors already, and don't want to increase this PR with that too.
Please remove the change request, probably we should open a new issue about it.